pkgs/common-updater/scripts/mark-broken

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env bash
+set -e
+
+scriptName=mark-broken # do not use the .wrapped name
+
+die() {
+    echo "$scriptName: error: $1" >&2
+    exit 1
+}
+
+usage() {
+    echo "Usage: $scriptName <attr> [--new-value=<new-value>]"
+}
+
+args=()
+
+for arg in "$@"; do
+    case $arg in
+        --new-value=*)
+            newValue="${arg#*=}"
+        ;;
+        --help)
+            usage
+            exit 0
+        ;;
+        --*)
+            echo "$scriptName: Unknown argument: $arg"
+            usage
+            exit 1
+        ;;
+        *)
+            args["${#args[*]}"]=$arg
+        ;;
+    esac
+done
+
+attr=${args[0]}
+
+if (( "${#args[*]}" < 1 )); then
+    echo "$scriptName: Too few arguments"
+    usage
+    exit 1
+fi
+
+if (( "${#args[*]}" > 1 )); then
+    echo "$scriptName: Too many arguments"
+    usage
+    exit 1
+fi
+
+if [ -z $newValue ]; then
+  newValue="true"
+fi
+
+nixFile=$(nix-instantiate --eval --json -E "with import ./. {}; (builtins.unsafeGetAttrPos \"description\" $attr.meta).file" | jq -r .)
+if [[ ! -f "$nixFile" ]]; then
+    die "Couldn't evaluate 'builtins.unsafeGetAttrPos \"description\" $attr.meta' to locate the .nix file!"
+fi
+
+# Insert broken attribute
+sed -i.bak "$nixFile" -r \
+  -e "/^\s*broken\s*=.*$/d" \
+  -e "s/(\s*)meta\s*=.*\{/&\n\1  broken = $newValue;/"
+
+if cmp -s "$nixFile" "$nixFile.bak"; then
+    mv "$nixFile.bak" "$nixFile"
+    die "Failed to mark the package as broken! Does it have a meta attribute?"
+fi
+
+if [[ "$newValue" == "true" ]]; then
+    # broken should evaluate to true in any case now
+    markedSuccessfully=$(nix-instantiate --eval -E "with import ./. {}; $attr.meta.broken" || true)
+    if [[ ! "$markedSuccessfully" == "true" ]]; then
```

Comment:
Either the check needs to be changed, or `markedSuccessfully` should be `|| "true"`. Also, maybe want to redirect stderr to /dev/null, or else you'll see `error: attribute 'broken' missing, at (string):1:21`.

```
$ if [[ ! $(true) == "true" ]]; then echo not-eq; fi
not-eq
```

Reply:
Again the error being printed in case of failure is intended. And I don't really get the other part, can you try to phrase your suggestion differently?

Reply:
Also fair.

My second point is about the fact that `markedSuccessfully` is either the string `true` if successfull, or the output of the command `true` if `nix-instantiate` fails (oh, maybe this is just so that the script doesn't exit because of `set -e`...). The comment above (`broken should evaluate to true in any case now`) led me to believe that if the instantiate errors, it should default to the string `true`, not the output of the command `true` (which is nothing and fails the check).

(Basically: ignore this, I wasn't thinking things through.)

Reply:
Yeah it's just because of `set -e`. The `true` command will never print "true" to stdout, so we should be fine.